### ISSUE TITLE
broken with ejs above 1.0.0

### ISSUE THREAD

**Issue by diffsky** [diffsky/ejstpl#1]
**reported from email**

My name is Sergio.y am a user Node module called ejstpl in my project. A few weeks ago with uglify working properly, but today I updated the module and when I use uglify get the following error:
 
Running "uglify:dist_theme" (uglify) task
{ message: '\'return\' outside of function',

I think .ejs files are not generated correctly and do not know what the reason. The .js generated from a .ejs is attached to the mail.

For installation of EJS pacakge.json I use the following:

```
"devDependencies": {
        "typescript": "~1.3.0",
        "less": "~2.1.1",
        "ejstpl":"~1.0.0",
        "typedoc":"*",
        "grunt": "~0.4.5",
        "grunt-contrib-clean":"*",
        "grunt-contrib-copy":"*",
        "grunt-contrib-imagemin":"*",
        "grunt-contrib-uglify": "~0.4.0",
        "grunt-contrib-cssmin": "~0.10.0",
        "grunt-mkdir":"*",
        "grunt-typescript": "~0.4.4",
        "grunt-html-build":"*",
        "grunt-shell":"*",
        "grunt-html-minify":"*",
        "grunt-text-replace":"*",
        "grunt-json-minify":"*"
    }
```

The command I use to generate js is:

```
ejstpl -c C:\Users\sergiolj\Proyectos\art\mobile/src/themes/fff/views/ -o src/themes/fff/views/ -i *.ejs
```

Thanks you very much

**bet.js**
```javascript
ejs=window.ejs||{tplc:{}};
ejs.tplc['bet'] = function(locals, escape, include, rethrow) {
  var lines = str.split('\n')
    , start = Math.max(lineno - 3, 0)
    , end = Math.min(lines.length, lineno + 3);

  // Error context
  var context = lines.slice(start, end).map(function(line, i){
    var curr = i + start + 1;
    return (curr == lineno ? ' >> ' : '    ')
      + curr
      + '| '
      + line;
  }).join('\n');

  // Alter exception message
  err.path = filename;
  err.message = (filename || 'ejs') + ':'
    + lineno + '\n'
    + context + '\n\n'
    + err.message;

  throw err;
};
escape = escape || ejs.escape;
var __output = ""; with (locals || {}) { ;__output += "<div class=\"header\" data-role=\"header\" data-position=\"fixed\">\n    <span class=\"open left\"><a href=\"#menu_left_panel\">&#61641;</a></span>\n    <span class=\"title\"><img class=\"title-image\" src=\"http://placehold.it/100x40\" alt=\"\"/></span>\n    <span class=\"open right\"><button class=\"btn button button_loguin\">";;__line = 4;;__output += escape(Globalization.getText("HEADER_BUTTON_LOGIN"));;__output += "</button></span>\n</div>\n\n<div class=\"content\" data-role=\"content\" class=\"scrollable\">\n    <div id=\"bets-time\">\n        <div id=\"bets-time-betContdown\">00:00</div>\n        <div><span id=\"text-idleague\">";;__line = 10;;__output += escape(Globalization.getText('BET_HEADER_TAG_LEAGUE'));;__output += "</span> <span id=\"idleague\">0</span> <span id=\"leagueWeek\">";;__output += escape(Globalization.getText('BET_HEADER_TAG_WEEK'));;__output += " <span id=\"leagueWeekNumber\">21</span></span></div>\n    </div>\n\n    <div data-role=\"tabs\" id=\"tabs\">\n        <div data-role=\"navbar\">\n            <ul>\n                <li><a id=\"id-tab-match-result\" href=\"#tab_id_matchResults\" data-ajax=\"false\" data-active=\"true\">";;__line = 16;;__output += escape(Globalization.getText('BET_TAB_MATCHRESULTS_TITLE'));;__output += "</a></li>\n                <li><a id=\"id-tab-match-golnogol\" href=\"#tab_id_gg_ng\" data-ajax=\"false\">";;__line = 17;;__output += escape(Globalization.getText('BET_TAB_GGNG_TITLE'));;__output += "</a></li>\n                <li><a id=\"id-tab-match-over-under\" href=\"#tab_id_over_under\" data-ajax=\"false\">";;__line = 18;;__output += escape(Globalization.getText('BET_TAB_OVERUNDER_TITLE'));;__output += "</a></li>\n            </ul>\n        </div>\n\n        <div id=\"tab_id_matchResults\" class=\"ui-body-d ui-content\">\n            <div>\n                <table style=\"width:100%;border: 1px solid #dddddd;background-color:#dddddd; line-height:0.75em; font-size:0.85em\">\n                    <tr>\n                        <td style=\"width:40%; text-align:center\"></td>\n                        <td style=\"width:20%; text-align:center\">";;__line = 27;;__output += escape(Globalization.getText('BET_TAB_MATCHRESULTS_HEADER_HOME'));;__output += "</td>\n                        <td style=\"width:20%; text-align:center\">";;__line = 28;;__output += escape(Globalization.getText('BET_TAB_MATCHRESULTS_HEADER_DRAW'));;__output += "</td>\n                        <td style=\"width:20%; text-align:center\">";;__line = 29;;__output += escape(Globalization.getText('BET_TAB_MATCHRESULTS_HEADER_AWAY'));;__output += "</td>\n                    </tr>\n                </table>\n            </div>\n            <ul id=\"list-view-results-matches\" data-role=\"listview\"></ul>\n        </div>\n\n        <div id=\"tab_id_gg_ng\">\n            <div>\n                <table style=\"width:100%;border: 1px solid #dddddd;background-color:#dddddd; line-height:0.75em; font-size:0.85em\">\n                    <tr>\n                        <td style=\"width:40%; text-align:center\"></td>\n                        <td style=\"width:20%; text-align:center\">Gol Gol</td>\n                        <td style=\"width:20%; text-align:center\">No Gol</td>\n                    </tr>\n                </table>\n            </div>\n            <ul id=\"list-view-gg-ng\" data-role=\"listview\"></ul>\n        </div>\n\n        <div id=\"tab_id_over_under\">\n            <div>\n                <table style=\"width:100%;border: 1px solid #dddddd;background-color:#dddddd; line-height:0.75em; font-size:0.85em\">\n                    <tr>\n                        <td style=\"width:40%; text-align:center\"></td>\n                        <td style=\"width:20%; text-align:center\">Over</td>\n                        <td style=\"width:20%; text-align:center\">Under</td>\n                    </tr>\n                </table>\n            </div>\n            <ul id=\"list-view-over-under\" data-role=\"listview\"></ul>\n        </div>\n\n        <div id=\"div_container_form_bet\">\n            <div id=\"div_button_market\">\n                <table style=\"width:100%\" >\n                    <tr>\n                        <td style=\"width:50%\">\n                            <div class=\"form-group has-feedback\" style=\"text-align:left;padding:5px;\">";;__line = 67;;__output += escape(Globalization.getText('BET_TAB_MATCHRESULTS_FOOTER_TAG_CHOOSEBETTYPE'));;__output += " (<span id=\"tag-bet-type\">NO Bet</span>)</div>\n                        </td>\n                        <td style=\"width:50%\">\n                            <div class=\"form-group has-feedback\" style=\"text-align:right;padding:5px\"><span id=\"bet_matchResult_counterSelections\">0</span> ";;__line = 70;;__output += escape(Globalization.getText('BET_TAB_MATCHRESULTS_FOOTER_TAG_SELECTION'));;__output += "</div>\n                        </td>\n                    </tr>\n                </table>\n                <div id=\"div_button_market_buttons\">\n                    <table style=\"width:100%;padding-top:3px;padding-bottom:3px\"><tr><td style=\"width:33%;text-align:left;padding-left: 5px\">G/C</td><td style=\"width:33%;text-align:center\">Amount</td><td style=\"width:33%;text-align:right; padding-right: 5px\">Max. Win</td></tr></table>\n                    <fieldset id=\"div_button_market_buttons_fieldset\" data-role=\"controlgroup\"  data-type=\"vertical\"></fieldset>\n                </div>\n            </div>\n            <div id=\"div_form_bets\">\n                <table style=\"width:100%\">\n                    <tr>\n                        <td style=\"width:50%\">\n                            <div class=\"form-group has-feedback\" style=\"text-align:left;padding:10px\">\n                                <label for=\"inputAmount\">";;__line = 84;;__output += escape(Globalization.getText('BET_TAB_MATCHRESULTS_FOOTER_TAG_STAKE'));;__output += ": </label>\n                                <input class=\"form-control\" id=\"inputAmount\" step=\"0.01\" placeholder=\"0.00 €\"/>\n                            </div>\n                        </td>\n                        <td style=\"width:50%\">\n                            <div id=\"\" style=\"text-align:right;padding:10px\">\n                                <label id=\"label-total-amount-win\" for=\"button_bet\">";;__line = 90;;__output += escape(Globalization.getText('BET_TAB_MATCHRESULTS_FOOTER_TAG_WIN'));;__output += ": <span id=\"bet_total_amount_win\">0</span> ";;__output += escape(Globalization.getCurrencyString());;__output += "</label>\n                                <button class=\"btn button\" id=\"button_bet\">";;__line = 91;;__output += escape(Globalization.getText('BET_TAB_MATCHRESULTS_FOOTER_BUTTON_BETNOW'));;__output += "</button>\n                            </div>\n                        </td>\n                    </tr>\n                </table>\n            </div>\n        </div>\n\n    </div>\n</div>";;__line = 100;}return __output.trim();
}
```

**Comment by diffsky** [comment]
It looks like things are breaking due to a newer release of the ejs package. Ejstpl pulls in the latest ejs version on install. I see some commits to ejs here:
https://github.com/tj/ejs/commits/master

If I revert ejstpl to use exactly ejs@1.0.0 tests pass ok, if I use latest, then tests fail. It's not clear to me exactly what changes have been made and I don't actively use ejs templating much these days so I won't be able to dig into it.

I'll publish a version of ejstpl that is locked to ejs@1.0.0.

**Comment by diffsky** [comment]
```
$ npm publish
+ ejstpl@3.4.0
```

**Comment by diffsky** [comment]
Looks like ejs above `1.0.0` has breaking changes:

> Version 2 of EJS makes some breaking changes with this version

https://github.com/tj/ejs/commit/2e4071f747cf5953b939b1939a84709ad4b86a47

**Comment by diffsky** [comment]
This is fixed in `ejstpl@4.0.0` which now works with ejs2 (but is not backwards compatible to ver 1).
